packages/dbml-core/src/parse/mysql/parser.pegjs

```diff
@@ -199,9 +199,9 @@ IndexInLineSyntax = _ unique:index_in_line
 	if(unique) {
 		index.unique = true;
 	}
-	const type = type2 || type1;
-	if(type)
-		index.type = type;
+	if(type1) index.type = type1;
+	else if(type2 && type2[1] && type2[1].type === 'index_type' && type2[1].value ) index.type = type2[1].value;
+	else if(type3) index.type = type3;
```

Comment:
```suggestion
        if (type2 && type2[1] && type2[1].type === 'index_type' && type2[1].value ) index.type = type2[1].value;
	else if (type1) index.type = type1;
```
Don't change the default behavior. If `IndexOption` is also an index_type will override the previous index_type.